Sources/SwiftProtobuf/Internal.swift

```diff
@@ -40,3 +40,32 @@ public struct Internal {
   }
 
 }
+
+// SwiftProtobuf Internal: Common support for storage classes to handle extension fields
+public protocol Internal_ExtMsgStorage: class {
+  associatedtype ExtendedMessage: Message
+  var extensionFieldValues: ExtensionFieldValueSet {get set}
+  func setExtensionValue<F: AnyExtensionField>(ext: MessageExtension<F, ExtendedMessage>, value: F.ValueType)
+  func getExtensionValue<F: AnyExtensionField>(ext: MessageExtension<F, ExtendedMessage>) -> F.ValueType
```

Comment:
Side note:  I don't think `setExtensionValue` and `getExtensionValue` need to be here.  The default implementations below should never, ever be overridden.